src/main/java/org/broadinstitute/consent/http/db/DataAccessRequestDAO.java

```diff
@@ -13,100 +15,165 @@
-            "          OR draft = true ) " +
-            "  AND ((data #>> '{}')::jsonb->>'userId')::numeric = :userId " +
-            "  ORDER BY ((data #>> '{}')::jsonb->>'sortDate')::numeric DESC")
-    List<DataAccessRequest> findAllDraftsByUserId(@Bind("userId") Integer userId);
+  /**
+   * Find DataAccessRequest by reference id
+   *
+   * @param referenceId String
+   * @return DataAccessRequest
+   */
+  @SqlQuery(
+      "SELECT id, reference_id, draft, user_id, create_date, sort_date, submission_date, update_date, (data #>> '{}')::jsonb AS data FROM data_access_request WHERE reference_id = :referenceId limit 1")
+  DataAccessRequest findByReferenceId(@Bind("referenceId") String referenceId);
 
-    /**
-     * Find DataAccessRequest by reference id
-     * @param referenceId String
-     * @return DataAccessRequest
-     */
-    @SqlQuery("SELECT id, reference_id, draft, (data #>> '{}')::jsonb AS data FROM data_access_request WHERE reference_id = :referenceId limit 1")
-    DataAccessRequest findByReferenceId(@Bind("referenceId") String referenceId);
+  /**
+   * Find DataAccessRequests by reference ids
+   *
+   * @param referenceIds List of Strings
+   * @return List<DataAccessRequest>
+   */
+  @SqlQuery(
+      "SELECT id, reference_id, draft, user_id, create_date, sort_date, submission_date, update_date, (data #>> '{}')::jsonb AS data FROM data_access_request WHERE reference_id IN (<referenceIds>)")
+  List<DataAccessRequest> findByReferenceIds(@BindList("referenceIds") List<String> referenceIds);
 
-    /**
-     * Find DataAccessRequests by reference ids
-     * @param referenceIds List of Strings
-     * @return List<DataAccessRequest>
-     */
-    @SqlQuery("SELECT id, reference_id, draft, (data #>> '{}')::jsonb AS data FROM data_access_request WHERE reference_id IN (<referenceIds>)")
-    List<DataAccessRequest> findByReferenceIds(@BindList("referenceIds") List<String> referenceIds);
+  /**
+   * Update DataAccessRequest by reference id and provided DataAccessRequestData
+   *
+   * @param referenceId String
+   * @param data DataAccessRequestData
+   */
+  @RegisterArgumentFactory(JsonArgumentFactory.class)
+  @SqlUpdate(
+      "UPDATE data_access_request SET data = to_jsonb(:data) WHERE reference_id = :referenceId")
+  void updateDataByReferenceId(
+      @Bind("referenceId") String referenceId, @Bind("data") @Json DataAccessRequestData data);
 
-    /**
-     * Update DataAccessRequest by reference id and provided DataAccessRequestData
-     * @param referenceId String
-     * @param data DataAccessRequestData
-     */
-    @RegisterArgumentFactory(JsonArgumentFactory.class)
-    @SqlUpdate("UPDATE data_access_request SET data = to_jsonb(:data) WHERE reference_id = :referenceId")
-    void updateDataByReferenceId(@Bind("referenceId") String referenceId, @Bind("data") @Json DataAccessRequestData data);
+  /**
+   * Update DataAccessRequest by reference id and provided DataAccessRequestData
```

Comment:
This is me nitpicking a bit, but could you expand on this comment a bit so it distinguishes the difference between this and the version 1 variant?

Reply:
I have another question. If we're updating a pre-existing record, shouldn't we fetch the original record, update the attributes individually with the incoming payload (like Object.assign or _.assign in JS), and then save the updated record? Right now the endpoint just takes the incoming json data and assigns it to the record, so if the payload is just ```{projectTitle: <string>}```, it can potentially update a fully filled DAR to a DAR that only has a projectTitle assigned.

Reply:
With a `PUT` request, that's the intention. What you're describing here is `PATCH`.

Reply:
Right, shouldn't this be a PATCH request?

Reply:
We don't currently support that method. We could support that, but it's not what the UI is expecting right now (it does a PUT). PATCH should be a separate story.

Reply:
Will file a story so we can look into this more but we need to be restrictive. Many fields on submitted DARs should not be updatable (i.e. datasetIds, etc.).

Reply:
See https://broadinstitute.atlassian.net/browse/DUOS-734 for this story.